heap/heap.py

```diff
@@ -1,21 +1,68 @@
+import math
 class Heap:
   def __init__(self):
     self.storage = []
 
   def insert(self, value):
-    pass
+    heap = self.storage
+    heap.append(value)
+    parent = math.floor((heap.index(value)-1)/2)
+    while heap[parent] < value:
+      return self._bubble_up(parent)
 
   def delete(self):
-    pass
+    heap = self.storage
+    toDelete = heap[0]
+    if len(heap) > 1:
+      heap[0] = heap[-1]
+      del heap[-1]
+      if len(heap) == 1:
+        return toDelete
+      elif len(heap) == 2:
+        if heap[0] < heap[1]:
+          replace = heap[0]
+          heap[0] = heap[1]
+          heap[1] = replace
+      elif heap[0] < heap[1] or heap[0] < heap[2]:
+        self._sift_down(0)
+    else:
+      heap = []
+    return toDelete
```

Comment:
I have to parse this code whenI'm more awake, but there might be logic here that should be discrete or be put in `_sift_down()`.

Yeah my 🧠needs a little break. 💦